### PR TITLE
implement GetEnteringArcs.

### PR DESCRIPTION
**Pull request by csukuangfj**
```cpp
// CHECK(CheckProperties(fsa, KTopSorted));
```
is currently disabled.

Should we use `glog` or implement our own `glog` like functions ?

### CODE REVIEW

**Comment by danpovey** [comment]
Fantastic!
I'm OK to include glog, depending what the others think...

**Comment by csukuangfj** [comment]
@qindazhu  @galv
what do you think about `glog`?

File: k2/csrc/fsa_util.cc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+#include <utility>
+#include <vector>
+
+namespace k2 {
+
+void GetEnteringArcs(const Fsa &fsa, VecOfVec *entering_arcs) {
+  // CHECK(CheckProperties(fsa, KTopSorted));
+
+  int num_states = fsa.NumStates();
```
Comment (qindazhu) [inline k2/csrc/fsa_util.cc:17]: it should be int32_t or StateId.

File: k2/csrc/fsa_util.cc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  std::vector<std::vector<std::pair<Label, StateId>>> vec(num_states);
+  int num_arcs = 0;
```
Comment (qindazhu) [inline k2/csrc/fsa_util.cc:19]: int32_t

Comment (csukuangfj) [reply k2/csrc/fsa_util.cc:19]: thanks, will change it in the next commit.

**Comment by qindazhu** [comment]
I'm ok with `glog`.

File: k2/csrc/fsa_util.cc (hunk 1 above)
Comment (csukuangfj) [reply k2/csrc/fsa_util.cc:17]: Fixed.

File: k2/csrc/fsa_util.cc (hunk 2 above)
Comment (csukuangfj) [reply k2/csrc/fsa_util.cc:19]: Done.

**Comment by galv** [comment]
glog-like functions are annoying to implement. We did that for kaldi, as people here are aware. Implementing KALDI_ERROR macro is actually pretty complicated because it requires throwing an exception in a destructor.

Also, glog, uses poor-ly named space macros like LOG, ERROR, etc.,so you may as well just use glog, in case another library also depends on it.

So I think glog is fine.

**Comment by csukuangfj** [comment]
@galv @qindazhu 
thanks, I'll include `glog` in later commits.

**Pull request merged**